code/game/atom/_atom.dm

```diff
@@ -861,7 +863,7 @@
 	var/extra_lines = 0
 	var/extra_context = ""
 
-	if (isliving(user) || isovermind(user) || isaicamera(user))
+	if(isliving(user) || isovermind(user) || isaicamera(user) || (isobserver(user) && ghost_screentips))
```

Comment:
Check `ghost_screentips` first. This is one of the most expensive procs in the entire game and so putting something expensive like `isobserver` first (O(n)) will have noticeable losses compared to the simple bool check.